tests/infra_test.go

```diff
@@ -212,6 +273,35 @@ func getNewLeaderPod(virtClient kubecli.KubevirtClient) *k8sv1.Pod {
 	return nil
 }
 
+func getKubevirtVMMetrics(virtClient kubecli.KubevirtClient, pod *k8sv1.Pod, containerName string) []string {
+	stdout, _, err := tests.ExecuteCommandOnPodV2(virtClient,
+		pod, containerName,
+		[]string{
+			"curl",
+			"-L",
+			"-k",
+			fmt.Sprintf("https://%s:%s/metrics", pod.Status.PodIP, "8443"),
+		})
+	Expect(err).ToNot(HaveOccurred())
+	return filterMetricsOutput(stdout, "kubevirt")
+}
+
+func parseMetricsToMap(lines []string) (map[string]float64, error) {
+	metrics := make(map[string]float64)
+	for _, line := range lines {
+		items := strings.Split(line, " ")
+		if len(items) != 2 {
+			return nil, fmt.Errorf("can't split properly line '%s'", line)
+		}
+		v, err := strconv.ParseFloat(items[1], 64)
```

Comment:
Can this fail because of parsing a comment line?

Reply:
Good point. Let me check this - also I'll try to add a test for it

Reply:
@rmohr we sidestep this issue thanks to filterMetricsOutput (see below https://github.com/kubevirt/kubevirt/pull/2077/files/262a8dee9be2f08ba27adfcd2d5583513906b73c#diff-29539fc2ce2beadde4ecbf90b4a00816L215), so the input of this function is a slice of lines which start with "kubevirt_" (regex equivalent would be "^kubevirt"). No comments could get there.
I'll push a small patch to rename filterMetricsOutput and to move it in the calling sites, to make this more evident. In a followup PR I'd like to extract this code and move it in a small helper package under `pkg/util`, with unit tests. How this plan sound?

Reply:
@fromanirh making it a little bit more evident that the output is already filtered is enough for me. You don't have to factor it out to pkg/util if you don't need it in the normal code from my point of view.

Reply:
@rmohr ack, on it.